src/lp_data/HighsLpUtils.cpp

```diff
@@ -2286,6 +2286,9 @@ HighsStatus readBasisStream(const HighsLogOptions& log_options,
 HighsStatus calculateColDuals(const HighsLp& lp, HighsSolution& solution) {
   //  assert(solution.row_dual.size() > 0);
   if (int(solution.row_dual.size()) < lp.num_row_) return HighsStatus::kError;
+  const bool is_colwise = lp.a_matrix_.isColwise();
+  assert(is_colwise);
```

Comment:
I don't understand this combination of assert and the next line. If this throws an assert, then the `kError` is unreachable?

Reply:
The `kError` return is there to ensure that a row-wise matrix isn't used in `release`. 
It shouldn't be necessary, as `calculateRowValues` and `calculateColDuals` are only called internally, so the assert in development should be enough.

Reply:
D'oh, silly me. The asserts aren't compiled into the `Release`.